docs/generate-model-from-third-party-resources.md

```diff
@@ -0,0 +1,139 @@
+# Guide to generate Java codes from CustomResourceDefinition
+
+__TL;DR__: This document will be useful when you extend third-party resources into your kubernetes cluster e.g. 
+[CustomResourceDefinition](https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions/)
+and try to program java to operate the extended APIs. The generation process requires your CRD to be defined with 
+[structral-schema](https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions/#specifying-a-structural-schema).
+
+
+### Steps
+
+
+##### Fetch openapi spec from the cluster
+
+Run the following command to fetch openapi spec from your existing cluster above 1.15+ version (with CRD already deployed):
+
+```bash
+kubectl get --raw="/openapi/v2" > /tmp/swagger # or any other file-path
+```
+
+Note that if you can't find a 1.15+ cluster anywhere, consider run one locally by [kind](https://github.com/bsycorp/kind) then
+don't forget to deploy your CRDs into the local-run kubernetes cluster.
+
+
+##### Generate Java model from the downloaded openapi spec
+
+Run the following command and there will be a java sources generated under `/tmp/java`. 
+
+```
+docker run -i --rm yue9944882/java-model-gen < /tmp/swagger | tar -xzf - -C /tmp/
```

Comment:
i don't have access to the offical registry so i pushed them my registry temporarily

Reply:
Neither do I (afaik)

Reply:
Is there a reference `Dockerfile` on how this image was generated?